pkg/kubernetes/scripts/virtual-machines/index.jsx

```diff
@@ -81,13 +82,17 @@ function addScopeVarsToStore ($scope) {
  * @param {kubeLoader} kubeLoader
  * @param {kubeSelect} kubeSelect
  * @param {kubeMethods} kubeMethods
+ * @param {KubeRequest} KubeRequest
  */
-function init($scope, kubeLoader, kubeSelect, kubeMethods) {
+function init($scope, kubeLoader, kubeSelect, kubeMethods, KubeRequest) {
     initReduxStore();
     addKubeLoaderListener($scope, kubeLoader, kubeSelect);
-    initMiddleware(kubeMethods, kubeLoader);
+    initMiddleware(kubeMethods, kubeLoader, KubeRequest);
     addScopeVarsToStore($scope);
 
+    watchMetrics(reduxStore);
```

Comment:
IIUC, this will cause active 5-seconds polling of node summary for all nodes where a VM is scheduled to.
No matter if stats are actually rendered for the user.
Considering the size of potential OC cluster, this does not scale.

If nothing better can be found, then at least next "polling-round" should not start until the previous one is fully finished plus some cool-down delay.

Or the polling logic can be changed to switch `on|off` when the UI needs to render the data.
This approach is more effective but looses history.

@martinpitt ?

Reply:
- next polling round is started after the last one is finished now

Should I poll it only when the tabs with metrics are active?

Reply:
The `kubernetes` plugin provides cluster-wide view. 
Means, all VMs of the cluster can be listed here and so for all cluster nodes the metrics are going to be polled. The metrics are rendered only if the `Usage` subtab of a VM is opened.

I'm afraid, this is highly ineffective causing permanent high data transfer and CPU load with 5 seconds cool-down windows.

Please consider the case of a cluster with hundreds of nodes and ten-thousands of VMs.

I still think we should start the polling pre-request.

Reply:
Ok, I will poll only opened vms. I think it is better for user experience  than just opened `Usage` subtab. It will make it more seamless when going through all the subtabs.

Reply:
ok, this makes sense.

Reply:
those in expanded state